### PR TITLE
feat: add date datatype

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> Allow date data type in context, traits, and properties 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by MoumitaM** [comment]
Created a task in the backlog to address vulnerabilities.

File: index.d.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ export interface apiObject {
     | boolean
     | undefined
     | apiObject
-    | (string | number | boolean | apiObject)[];
+    | Date
+    | (string | number | boolean | apiObject | Date)[];
```
Comment (saikumarrs) [inline index.d.ts:15]: Let's also support `null` and `undefined` values.

```suggestion
    | null
    | Date
    | (string | number | boolean | apiObject | Date | null | undefined)[];
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Suggested changes

File: index.d.ts (hunk 1 above)
Comment (MoumitaM) [reply index.d.ts]: Addressed

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**